(PDB-3207) Backport ezbake reload and logback file-sized based rotation

camlow325 opened the pull request:
This PR backports changes done originally for the PuppetDB 4.2.4 release for:

1) Adding a 'reload' action (via ezbake) that the service framework can use to
    HUP the puppetdb service.

2) Add file-size based log rotation to the logback configuration.

This PR includes cherry-picks of commits from the following PRs that were
previously merged for the 4.2.4 release: #2087, #2091, #2097, #2098, and
#2102.

camlow325 commented:
Putting this up for review but this shouldn't be merged until we have a CI pipeline running for the 4.2.3.x branch.  Having that pipeline in place is currently blocked on https://github.com/puppetlabs/ci-job-configs/pull/2110.

camlow325 commented:
Ping @cprice404 and @jpinsonault for review.

jpinsonault commented:
👍 looks good to me

puppetcla commented:
CLA signed by all contributors.

wkalt commented:
looks good to me. I'd be inclined to squash the commits but reasonable people may differ.

camlow325 commented:
Thanks, @wkalt.  I'd be good with squashing the commits.  I'll plan on doing that after we see if the CI pipeline for the new branch is working (assuming no one else has chimed in with a different opinion before I get to that).

puppetlabs-jenkins commented:
Test FAILed

camlow325 commented:
@wkalt - fyi, I rebased the PR.  Only change from before is that I deleted the commit that would have bumped beaker to an older version than what is now on the 4.2.3.x branch.

puppetlabs-jenkins commented:
Test PASSed
